rpm/0084-sailfishos-gecko-Fix-download-starting-when-multiple.patch

```diff
@@ -0,0 +1,48 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Andrew den Exter <andrew.den.exter@jolla.com>
+Date: Fri, 5 Nov 2021 08:55:22 +0000
+Subject: [PATCH] [sailfishos][gecko] Fix download starting when multiple tabs
+ are closed. JB#56047 OMP#JOLLA-476
+
+When the active tab is closed another tab will be made active and start
+to load, if that tab is closed while loading the parent puppet widget can be
+destroyed before the content view is created causing a failure in the
+content view construction the error from which gets interpreted as the
+unsupported document type and the browser falls back to downloading.
+
+There is a failure case for when this happens and the DocShell is in
+destroyed state which cancels the loading the document, so we can do the
+same if the puppet widget has been destroyed.
+---
+ docshell/base/nsDocShell.cpp | 8 ++++++--
+ 1 file changed, 6 insertions(+), 2 deletions(-)
+
+diff --git a/docshell/base/nsDocShell.cpp b/docshell/base/nsDocShell.cpp
+index ae8b77a80cdc..912ca0b67a5a 100644
+--- a/docshell/base/nsDocShell.cpp
++++ b/docshell/base/nsDocShell.cpp
+@@ -7659,7 +7659,8 @@ nsresult nsDocShell::CreateContentViewer(const nsACString& aContentType,
+     }
+   }
+ 
+-  NS_ENSURE_SUCCESS(Embed(viewer), NS_ERROR_FAILURE);
++  rv = Embed(viewer)
++  NS_ENSURE_SUCCESS(rv, rv);
```

Comment:
This looks like a void to me.

Reply:
I don't know what you mean.

The change itself is in the event of an error returning the error value from Embed() instead of NS_ERROR_FAILURE allowing the NS_ERROR_DOCSHELL_DYING from SetupNewViewer() to reach nsDSURIContentListener::DoContent() where the content download will be cancelled and reported as being successfully handled instead of returning a failure which then results in a fallback download handler for the content.